src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java

```diff
@@ -43,26 +48,57 @@ public NewHandler(IndentationCheck indentCheck,
 
     @Override
     public void checkIndentation() {
-        final DetailAST type = getMainAst().getFirstChild();
-        if (type != null) {
-            checkExpressionSubtree(type, getIndent(), false, false);
+        final DetailAST firstChild = getMainAst().getFirstChild();
+        final DetailAST mainAst = getMainAst();
```

Comment:
```
final DetailAST mainAst = getMainAst();
final DetailAST firstChild = mainAst.getFirstChild();
```

Reply:
Also, you can move initialization of firstChild lower, closer to its first usage